### ISSUE TITLE
[YSQL] YBDB using Diesel seg faults with LIMIT '1'

### ISSUE THREAD

**Issue by yb-andrew** [yugabyte/yugabyte-db#3811]
Was hooking up Rust's Diesel ORM with YugabyteDB and got a `server closed the connection unexpectedly` segmentation fault. Reason is most likely LIMIT 1 (where the number can be either a string or integer).
Log:
[yb-diesel-error.log](https://github.com/yugabyte/yugabyte-db/files/4278754/yb-diesel-error.log)

**Comment by m-iancu** [comment]
Looks like the issue is indeed the `LIMIT` clause.
Simplest way to replicate locally:
```
CREATE TABLE test(a int primary key);
SELECT * FROM "information_schema"."key_column_usage" LIMIT 1;
```
(generally, if the limit is less or equal to the total number of rows the backend crashes, if it is strictly larger than it works).
The stack trace is:

```
  * frame #0: 0x0000000100ff1e60 postgres`InstrEndLoop(instr=0x0000000000000000) at instrument.c:119:14 [opt]
    frame #1: 0x0000000101dc060c yb_pg_metrics.so`ybpgm_ExecutorEnd(queryDesc=0x0000000111276cd8) at yb_pg_metrics.c:465:2 [opt]
    frame #2: 0x0000000100ff110a postgres`ShutdownSQLFunction [inlined] postquel_end(es=<unavailable>) at functions.c:890:3 [opt]
    frame #3: 0x0000000100ff10ec postgres`ShutdownSQLFunction(arg=4410491152) at functions.c:1472 [opt]
    frame #4: 0x0000000100feec9f postgres`FreeExecutorState [inlined] ShutdownExprContext(econtext=0x0000000111636ea0) at execUtils.c:870:4 [opt]
    frame #5: 0x0000000100feec47 postgres`FreeExecutorState [inlined] FreeExprContext(econtext=0x0000000111636ea0) at execUtils.c:383 [opt]
    frame #6: 0x0000000100feec47 postgres`FreeExecutorState(estate=0x0000000111636110) at execUtils.c:214 [opt]
    frame #7: 0x0000000100fe4a73 postgres`standard_ExecutorEnd(queryDesc=0x0000000106c6f110) at execMain.c:512:2 [opt]
    frame #8: 0x0000000100f9d9b7 postgres`PortalCleanup(portal=<unavailable>) at portalcmds.c:300:4 [opt]
    frame #9: 0x00000001012a1a12 postgres`PortalDrop(portal=0x0000000106df4110, isTopCommit=<unavailable>) at portalmem.c:499:3 [opt]
    frame #10: 0x00000001011526e6 postgres`yb_exec_simple_query_attempting_to_restart_read at postgres.c:1169:3 [opt]
    frame #11: 0x0000000101152112 postgres`yb_exec_simple_query_attempting_to_restart_read(query_string="SELECT * FROM \"information_schema\".\"key_column_usage\" LIMIT 1;", exec_context=0x0000000106dba000) at postgres.c:4019 [opt]
    frame #12: 0x0000000101150bb7 postgres`PostgresMain(argc=1, argv=<unavailable>, dbname=<unavailable>, username="yugabyte") at postgres.c:0 [opt]
    frame #13: 0x00000001010ce976 postgres`BackendRun(port=<unavailable>) at postmaster.c:4411:2 [opt]
    frame #14: 0x00000001010cdfd3 postgres`ServerLoop [inlined] BackendStartup(port=<unavailable>) at postmaster.c:4077:3 [opt]
    frame #15: 0x00000001010cdfaf postgres`ServerLoop at postmaster.c:1737 [opt]
    frame #16: 0x00000001010ca7e6 postgres`PostmasterMain(argc=23, argv=<unavailable>) at postmaster.c:1400:11 [opt]
    frame #17: 0x0000000101035b1b postgres`PostgresServerProcessMain(argc=<unavailable>, argv=<unavailable>) at main.c:234:3 [opt]
    frame #18: 0x0000000101035d72 postgres`main + 34
    frame #19: 0x00007fff67df77fd libdyld.dylib`start + 1
```

**Comment by m-iancu** [comment]
Fixed by 9aed9fb.
